### ISSUE TITLE
Spark 3.2 support?

### ISSUE THREAD

**Issue by matschaffer-roblox** [LucaCanali/Miscellaneous#3]
Hi @LucaCanali - hope this is an okay channel to reach out on. Love your spark dashboards! 🧡

I was trying to set this up for work on AWS EMR and it _seems_ like the metrics listed on https://spark.apache.org/docs/latest/monitoring.html#component-instance--executor don't get produced from applications on spark prior to 3.3

But I see 3.2 listed in the tags of https://github.com/apache/spark/commit/1ffe03d9e87fb784cc8a0bae232c81c7b14deac9

Do you happen to know if there's a way to get your dashboards working on 3.2? Perhaps the metrics.properties just needs to be different?

I'm specially curious about the active jobs and executor run time per process graphs.

These work fine for me on 3.3

![Screen Shot 2022-11-23 at 7 49 30 PM](https://user-images.githubusercontent.com/115565899/203689835-e13557d2-9bbf-4949-b272-b6aae4095c80.png)

But if I set up an EMR running spark 3.2 to publish to the same influx, it's just blank.

![Screen Shot 2022-11-23 at 7 51 13 PM](https://user-images.githubusercontent.com/115565899/203689997-b0822f02-47b1-4aca-9ae1-7a742b6ae7a8.png)

**Comment by matschaffer-roblox** [comment]
Oh, maybe this isn't a 3.2/3.3 problem but something involving EMR. When running spark locally everything lights up fine, but if I run 3.3 in EMR I get the same missing Executor metrics.

![Screen Shot 2022-11-24 at 12 12 43 PM](https://user-images.githubusercontent.com/115565899/203858298-58441c29-ef8c-4f45-9ab0-1ba6bf33def1.png)

**Comment by LucaCanali** [comment]
Hi @matschaffer-roblox I am glad you like the Spark dashboard and that you could make it work for your environment with Spark 3.3.
I regularly use the dashboard with Spark 3.2 so I guess the issue you observe may rather be in the configuration.

I am trying to guess what could be the problem and suggest a test: I believe in Spark 3.2 you may need to revert to use configuration parameters (--conf "spark.metrics.conf.....") rather than using metrics.conf in certain envronments, notably on k8s. Can you please try testing that?

Another important point is that the executors need to be able to send the metrics to the InfluxDB instance, as configured with spark.metrics.conf.sink.graphite.host=<host_with_InfluxDB> so you want to double check that there are no firewall/network issues on that too.

Best,  
Luca

**Comment by matschaffer-roblox** [comment]
Thanks so much for the reply Luca! And glad to hear you are successfully using this on 3.2 as well. So this is likely some EMR-specific problem I'm hitting here.

Your suggestion is spot-on. As a test I've started using this:

```
spark-shell \
    --conf "spark.metrics.conf.*.sink.csv.class=org.apache.spark.metrics.sink.CsvSink" \
    --conf "spark.metrics.conf.*.sink.csv.directory=$(pwd)/metrics"
```

Which makes it quite easy to confirm the presence (or absence in EMR) of the `driver.executor` metric space.

```
[hadoop@ip-... ~]$ ls metrics/*numEventsPosted**
metrics/application_1669323372614_0001.driver.LiveListenerBus.numEventsPosted.csv
[hadoop@ip-... ~]$ ls metrics/*bytesRead*
ls: cannot access metrics/*bytesRead*: No such file or directory
[hadoop@ip-... ~]$ ls metrics/*cpuTime*
ls: cannot access metrics/*cpuTime*: No such file or directory
[hadoop@ip-... ~]$ ls metrics/*runTime*
ls: cannot access metrics/*runTime*: No such file or directory
```

Or on local:

```
❯ ls metrics/*bytesRead*
metrics/local-1669322497429.driver.executor.bytesRead.csv
```

I've opened a support issue with AWS to see if they have any idea what might be suppressing metrics from that part of the code.

I think we can probably close this issue then. Thanks again for the cross-check! Hopefully I can figure something out with AWS and get this dashboard going for our production workloads soon 🤞

**Comment by LucaCanali** [comment]
Good to hear that using configuration parameters worked for you on Spark 3.2. In that case you can consider also setting the configuration parameters in spark-defaults.conf.

**Comment by matschaffer-roblox** [comment]
Yep! Those seem to work. Just need to now find out why those executor metrics aren't getting sent on EMR. Thanks again for the advice!

**Comment by matschaffer-roblox** [comment]
Thought you might like to hear I got this worked out. I'm still learning spark so wasn't aware that the executor metrics are only emitted by EMR's core or task node JVMs. Once I instrumented the whole cluster, things started coming through.

I've managed to get your dashboard mostly transliterated over to our promql-based telemetry infrastructure. 

Thanks again for all your work here. It made bootstrapping our spark observability efforts a _lot_ easier!
